### PR TITLE
Feat/api sec azure

### PR DESCRIPTION
**Pull request by HomelessDinosaur**

### CODE REVIEW

File: pkg/provider/pulumi/azure/subscriptions.go [hunk 1]
```diff
@@ -106,7 +106,9 @@ func newSubscriptions(ctx *pulumi.Context, name string, args *SubscriptionsArgs,
 				WebhookEndpoint: pulumiEventgrid.EventSubscriptionWebhookEndpointArgs{
 					Url: hostUrl,
 					// TODO: Reduce event chattiness here and handle internally in the Azure AppService HTTP Gateway?
-					MaxEventsPerBatch: pulumi.Int(1),
+					MaxEventsPerBatch:         pulumi.Int(1),
+					ActiveDirectoryAppIdOrUri: app.Sp.ClientID,
+					ActiveDirectoryTenantId:   app.Sp.TenantID,
```
Comment (tjholm) [inline pkg/provider/pulumi/azure/subscriptions.go:111]: @HomelessDinosaur Did this end up working or were there still issues with auth for events?

Comment (HomelessDinosaur) [reply pkg/provider/pulumi/azure/subscriptions.go:111]: The auth for events works, as does the API auth.

File: pkg/provider/pulumi/azure/serviceprinciple.go [hunk 2]
```diff
@@ -40,10 +40,30 @@ func newServicePrincipal(ctx *pulumi.Context, name string, args *ServicePrincipa
 	if err != nil {
 		return nil, err
 	}
+	current, err := azuread.GetClientConfig(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
 
 	// create an application per service principal
+	appRoleId := pulumi.String("4962773b-9cdb-44cf-a8bf-237846a00ab7")
```
Comment (davemooreuws) [inline pkg/provider/pulumi/azure/serviceprinciple.go:49]: Is it better that this (line 64 as well) is a new uuid each time?

**Review by davemooreuws — COMMENTED** [review]
Hey just one comment, looking good.

File: pkg/provider/pulumi/azure/serviceprinciple.go (hunk 2 above)
Comment (HomelessDinosaur) [reply pkg/provider/pulumi/azure/serviceprinciple.go:49]: From my understanding, that's a specific uuid which points to the `AzureEventGridSecureWebhookSubscriber` Role

Comment (davemooreuws) [reply pkg/provider/pulumi/azure/serviceprinciple.go:49]: I see, sounds good!

**Review by davemooreuws — APPROVED** [review]

**Review by tjholm — APPROVED** [review]

**Pull request merged**